--- tb_null.tcl.orig	2020-05-22 14:23:40.028403800 +0200
+++ tb_null.tcl	2020-05-23 17:05:15.283753700 +0200
@@ -9,3 +9,7 @@
 }
 
+add_force {/tb_null/msys_wrapper_i/msys_i/SC0712_0/reset_out} -radix hex {1 10us}
+add_force {/tb_null/msys_wrapper_i/FFT_controller_i/rx09_bs_32bits_vld} -radix hex {1 10us} {0 10010000000fs} -repeat_every 50000000fs
+add_force {/tb_null/msys_wrapper_i/FFT_controller_i/rx24_bs_32bits_vld} -radix hex {1 10us} {0 10010000000fs} -repeat_every 50000000fs
+
 run 1000ns
