--- msys_wrapper.vhd.old	2020-04-30 20:14:00.524470400 +0200
+++ msys_wrapper.vhd	2020-04-30 20:25:16.763814100 +0200
@@ -42,5 +42,5 @@
     ETH0_DA_G : out STD_LOGIC_VECTOR ( 0 to 0 );
     ETH0_DA_Y : out STD_LOGIC;
-    ETH0_LINK_LED_g : in STD_LOGIC;
+    ETH0_LINK_LED : in STD_LOGIC;
     ETH0_MDIO_MDC_mdc : out STD_LOGIC;
     ETH0_MDIO_MDC_mdio_io : inout STD_LOGIC;
@@ -226,4 +226,10 @@
   );
   end component IOBUF;
+  component IBUF is
+  port (
+    I : in STD_LOGIC;
+    O : out STD_LOGIC
+  );
+  end component IBUF;
   signal BOARD_IIC_scl_i : STD_LOGIC;
   signal BOARD_IIC_scl_o : STD_LOGIC;
@@ -235,4 +241,5 @@
   signal ETH0_MDIO_MDC_mdio_o : STD_LOGIC;
   signal ETH0_MDIO_MDC_mdio_t : STD_LOGIC;
+  signal ETH0_LINK_LED_g : STD_LOGIC;
   signal TRX_spi_io0_i : STD_LOGIC;
   signal TRX_spi_io0_o : STD_LOGIC;
@@ -317,4 +324,9 @@
       T => TRX_spi_ss_t
     );
+ETH0_LINK_LED_0: component IBUF
+     port map (
+      I => ETH0_LINK_LED,
+      O => ETH0_LINK_LED_g
+    );
 msys_i: component msys
      port map (
