--- msys_wrapper.vhd.orig	2020-05-13 10:43:22.571687800 +0200
+++ msys_wrapper.vhd	2020-05-13 10:05:49.604333700 +0200
@@ -2,5 +2,5 @@
 ----------------------------------------------------------------------------------
 --Tool Version: Vivado v.2019.2.1 (win64) Build 2729669 Thu Dec  5 04:49:17 MST 2019
---Date        : Wed May 13 10:43:22 2020
+--Date        : Tue May 12 23:32:05 2020
 --Host        : ULRICHHABEL6701 running 64-bit major release  (build 9200)
 --Command     : generate_target msys_wrapper.bd
@@ -16,4 +16,6 @@
     BOARD_IIC_scl_io : inout STD_LOGIC;
     BOARD_IIC_sda_io : inout STD_LOGIC;
+    BOARD_ROTENC_I : in STD_LOGIC;
+    BOARD_ROTENC_Q : in STD_LOGIC;
     BOARD_ROTENC_PUSH : in STD_LOGIC;
     CLK0_clk_n : in STD_LOGIC_VECTOR ( 0 to 0 );
@@ -42,5 +44,5 @@
     ETH0_DA_G : out STD_LOGIC_VECTOR ( 0 to 0 );
     ETH0_DA_Y : out STD_LOGIC;
-    ETH0_LINK_LED_g : in STD_LOGIC;
+    ETH0_LINK_LED : in STD_LOGIC;
     ETH0_MDIO_MDC_mdc : out STD_LOGIC;
     ETH0_MDIO_MDC_mdio_io : inout STD_LOGIC;
@@ -64,8 +66,6 @@
     TRX_reset : out STD_LOGIC_VECTOR ( 0 to 0 );
     TRX_rfx_mode : out STD_LOGIC_VECTOR ( 0 to 0 );
-    TRX_rx09_bs_i : in STD_LOGIC_VECTOR ( 31 downto 0 );
     TRX_rx09_fifo_o : out STD_LOGIC_VECTOR ( 31 downto 0 );
     TRX_rx09_fifo_valid_o : out STD_LOGIC;
-    TRX_rx24_bs_i : in STD_LOGIC_VECTOR ( 31 downto 0 );
     TRX_rx24_fifo_o : out STD_LOGIC_VECTOR ( 31 downto 0 );
     TRX_rx24_fifo_valid_o : out STD_LOGIC;
@@ -101,8 +101,5 @@
     qspi_flash_io3_io : inout STD_LOGIC;
     qspi_flash_ss_io : inout STD_LOGIC;
-    reset : in STD_LOGIC;
-    rotenc_dec_cnt_en : in STD_LOGIC;
-    rotenc_dec_cnt_up_dwn : in STD_LOGIC;
-    rotenc_decoder_reset : out STD_LOGIC_VECTOR ( 0 to 0 )
+    reset : in STD_LOGIC
   );
 end msys_wrapper;
@@ -230,4 +227,10 @@
   );
   end component msys;
+  component IBUF is
+  port (
+    I : in STD_LOGIC;
+    O : out STD_LOGIC
+  );
+  end component IBUF;
   component IOBUF is
   port (
@@ -238,4 +241,26 @@
   );
   end component IOBUF;
+  component rotenc_decoder is
+  port (
+    clk : in STD_LOGIC;
+    reset : in STD_LOGIC;
+    rotenc_I : in STD_LOGIC;
+    rotenc_Q : in STD_LOGIC;
+    cnt_up_dwn : out STD_LOGIC;
+    cnt_en : out STD_LOGIC
+  );
+  end component rotenc_decoder;
+  component auto_LVDS_rotate is
+  port (
+   resetn       : in  STD_LOGIC;
+   clk          : in  STD_LOGIC;
+   LVDS24       : in  STD_LOGIC_VECTOR (31 downto 0);
+   LVDS24_valid : in  STD_LOGIC;
+   LVDS09       : in  STD_LOGIC_VECTOR (31 downto 0);
+   LVDS09_valid : in  STD_LOGIC;
+   rot24q       : out STD_LOGIC_VECTOR (31 downto 0);
+   rot09q       : out STD_LOGIC_VECTOR (31 downto 0)
+  );
+  end component auto_LVDS_rotate;
   signal BOARD_IIC_scl_i : STD_LOGIC;
   signal BOARD_IIC_scl_o : STD_LOGIC;
@@ -244,4 +269,5 @@
   signal BOARD_IIC_sda_o : STD_LOGIC;
   signal BOARD_IIC_sda_t : STD_LOGIC;
+  signal ETH0_LINK_LED_g : STD_LOGIC;
   signal ETH0_MDIO_MDC_mdio_i : STD_LOGIC;
   signal ETH0_MDIO_MDC_mdio_o : STD_LOGIC;
@@ -260,4 +286,11 @@
   signal TRX_spi_ss_o_0 : STD_LOGIC_VECTOR ( 0 to 0 );
   signal TRX_spi_ss_t : STD_LOGIC;
+  signal TRX_rx09_fifo : STD_LOGIC_VECTOR ( 31 downto 0 );
+  signal TRX_rx24_fifo : STD_LOGIC_VECTOR ( 31 downto 0 );
+  signal TRX_rx09_fifo_valid : STD_LOGIC;
+  signal TRX_rx24_fifo_valid : STD_LOGIC;
+  signal TRX_rx09_bs : STD_LOGIC_VECTOR ( 31 downto 0 );
+  signal TRX_rx24_bs : STD_LOGIC_VECTOR ( 31 downto 0 );
+  signal microblaze_0_Clk_100MHz : STD_LOGIC;
   signal onewire_EUI48_tri_i_0 : STD_LOGIC_VECTOR ( 0 to 0 );
   signal onewire_EUI48_tri_io_0 : STD_LOGIC_VECTOR ( 0 to 0 );
@@ -279,5 +312,13 @@
   signal qspi_flash_ss_o : STD_LOGIC;
   signal qspi_flash_ss_t : STD_LOGIC;
+  signal rotenc_dec_cnt_en : STD_LOGIC;
+  signal rotenc_dec_cnt_up_dwn : STD_LOGIC;
+  signal rotenc_decoder_reset : STD_LOGIC;
 begin
+ETH0_LINK_LED_0: component IBUF
+     port map (
+      I => ETH0_LINK_LED,
+      O => ETH0_LINK_LED_g
+    );
 BOARD_IIC_scl_iobuf: component IOBUF
      port map (
@@ -329,4 +370,24 @@
       T => TRX_spi_ss_t
     );
+rotenc_decoder_i: component rotenc_decoder
+     port map (
+      reset => rotenc_decoder_reset,
+      clk => microblaze_0_Clk_100MHz,
+      rotenc_I => BOARD_ROTENC_I,
+      rotenc_Q => BOARD_ROTENC_Q,
+      cnt_up_dwn => rotenc_dec_cnt_up_dwn,
+      cnt_en => rotenc_dec_cnt_en
+    );
+auto_LVDS_rotate_i: component auto_LVDS_rotate
+     port map (
+      resetn => rotenc_decoder_reset,
+      clk => microblaze_0_Clk_100MHz,
+      LVDS24 => TRX_rx24_fifo,
+      LVDS24_valid => TRX_rx24_fifo_valid,
+      LVDS09 => TRX_rx09_fifo,
+      LVDS09_valid => TRX_rx09_fifo_valid,
+      rot24q => TRX_rx24_bs,
+      rot09q => TRX_rx09_bs
+    );
 msys_i: component msys
      port map (
@@ -387,10 +448,4 @@
       TRX_reset(0) => TRX_reset(0),
       TRX_rfx_mode(0) => TRX_rfx_mode(0),
-      TRX_rx09_bs_i(31 downto 0) => TRX_rx09_bs_i(31 downto 0),
-      TRX_rx09_fifo_o(31 downto 0) => TRX_rx09_fifo_o(31 downto 0),
-      TRX_rx09_fifo_valid_o => TRX_rx09_fifo_valid_o,
-      TRX_rx24_bs_i(31 downto 0) => TRX_rx24_bs_i(31 downto 0),
-      TRX_rx24_fifo_o(31 downto 0) => TRX_rx24_fifo_o(31 downto 0),
-      TRX_rx24_fifo_valid_o => TRX_rx24_fifo_valid_o,
       TRX_rx_clk_64MHz_clk_n => TRX_rx_clk_64MHz_clk_n,
       TRX_rx_clk_64MHz_clk_p => TRX_rx_clk_64MHz_clk_p,
@@ -413,4 +468,10 @@
       TRX_tx_data_n(1 downto 0) => TRX_tx_data_n(1 downto 0),
       TRX_tx_data_p(1 downto 0) => TRX_tx_data_p(1 downto 0),
+      TRX_rx09_fifo_o(31 downto 0) => TRX_rx09_fifo(31 downto 0),
+      TRX_rx24_fifo_o(31 downto 0) => TRX_rx24_fifo(31 downto 0),
+      TRX_rx09_fifo_valid_o => TRX_rx09_fifo_valid,
+      TRX_rx24_fifo_valid_o => TRX_rx24_fifo_valid,
+      TRX_rx09_bs_i => TRX_rx09_bs,
+      TRX_rx24_bs_i => TRX_rx24_bs,
       UART0EXT_CTSn(0) => UART0EXT_CTSn(0),
       UART0EXT_DCDn(0) => UART0EXT_DCDn(0),
@@ -424,5 +485,5 @@
       UART0_txd => UART0_txd,
       ULI_SYSTEM_XIO => ULI_SYSTEM_XIO,
-      microblaze_0_Clk_100MHz_o => microblaze_0_Clk_100MHz_o,
+      microblaze_0_Clk_100MHz_o => microblaze_0_Clk_100MHz,
       onewire_EUI48_tri_i(0) => onewire_EUI48_tri_i_0(0),
       onewire_EUI48_tri_o(0) => onewire_EUI48_tri_o_0(0),
@@ -447,5 +508,5 @@
       rotenc_dec_cnt_en => rotenc_dec_cnt_en,
       rotenc_dec_cnt_up_dwn => rotenc_dec_cnt_up_dwn,
-      rotenc_decoder_reset(0) => rotenc_decoder_reset(0)
+      rotenc_decoder_reset(0) => rotenc_decoder_reset
     );
 onewire_EUI48_tri_iobuf_0: component IOBUF
