--- msys_wrapper.vhd.orig	2020-04-29 12:08:13.008675400 +0200
+++ msys_wrapper.vhd	2020-04-29 12:28:07.247384300 +0200
@@ -17,4 +17,6 @@
     BOARD_IIC_sda_io : inout STD_LOGIC;
     BOARD_ROTENC_PUSH : in STD_LOGIC;
+    BOARD_ROTENC_I : in STD_LOGIC;
+    BOARD_ROTENC_Q : in STD_LOGIC;
     CLK0_clk_n : in STD_LOGIC_VECTOR ( 0 to 0 );
     CLK0_clk_p : in STD_LOGIC_VECTOR ( 0 to 0 );
@@ -94,9 +96,5 @@
     qspi_flash_io3_io : inout STD_LOGIC;
     qspi_flash_ss_io : inout STD_LOGIC;
-    reset : in STD_LOGIC;
-    rotenc_dec_cnt_en : in STD_LOGIC;
-    rotenc_dec_cnt_up_dwn : in STD_LOGIC;
-    rotenc_decoder_clk : out STD_LOGIC;
-    rotenc_decoder_reset : out STD_LOGIC_VECTOR ( 0 to 0 )
+    reset : in STD_LOGIC
   );
 end msys_wrapper;
@@ -226,4 +224,14 @@
   );
   end component IOBUF;
+  component rotenc_decoder is
+  port (
+    clk : in STD_LOGIC;
+    reset : in STD_LOGIC;
+    rotenc_I : in STD_LOGIC;
+    rotenc_Q : in STD_LOGIC;
+    cnt_up_dwn : out STD_LOGIC;
+    cnt_en : out STD_LOGIC
+  );
+  end component rotenc_decoder;
   signal BOARD_IIC_scl_i : STD_LOGIC;
   signal BOARD_IIC_scl_o : STD_LOGIC;
@@ -267,4 +275,8 @@
   signal qspi_flash_ss_o : STD_LOGIC;
   signal qspi_flash_ss_t : STD_LOGIC;
+  signal rotenc_dec_cnt_en : STD_LOGIC;
+  signal rotenc_dec_cnt_up_dwn : STD_LOGIC;
+  signal rotenc_decoder_clk : STD_LOGIC;
+  signal rotenc_decoder_reset : STD_LOGIC;
 begin
 BOARD_IIC_scl_iobuf: component IOBUF
@@ -317,4 +329,13 @@
       T => TRX_spi_ss_t
     );
+rotenc_decoder_i: component rotenc_decoder
+     port map (
+      clk => rotenc_decoder_clk,
+      reset => rotenc_decoder_reset,
+      rotenc_I => BOARD_ROTENC_I,
+      rotenc_Q => BOARD_ROTENC_Q,
+      cnt_up_dwn => rotenc_dec_cnt_up_dwn,
+      cnt_en => rotenc_dec_cnt_en
+    );
 msys_i: component msys
      port map (
@@ -429,5 +450,5 @@
       rotenc_dec_cnt_up_dwn => rotenc_dec_cnt_up_dwn,
       rotenc_decoder_clk => rotenc_decoder_clk,
-      rotenc_decoder_reset(0) => rotenc_decoder_reset(0)
+      rotenc_decoder_reset(0) => rotenc_decoder_reset
     );
 onewire_EUI48_tri_iobuf_0: component IOBUF
